src/Diagnostics.RuntimeHost/Controllers/ProcessHealthController.cs

```diff
@@ -13,18 +15,44 @@ public class ProcessHealthController : Controller
     {
         private ISourceWatcherService _sourceWatcherService;
         private ICompilerHostClient _compilerHostClient;
+        private IConfiguration _configuration;
 
-        public ProcessHealthController(ISourceWatcherService sourceWatcherService, ICompilerHostClient compilerHostClient)
+        public ProcessHealthController(ISourceWatcherService sourceWatcherService, ICompilerHostClient compilerHostClient, IConfiguration Configuration)
         {
             // These dependencies are injected for the services to start.
             _sourceWatcherService = sourceWatcherService;
             _compilerHostClient = compilerHostClient;
+            _configuration = Configuration;
+        }
+
+        public static bool PingExternal(string pingUrl)
+        {
+            try
+            {
+                using (var client = new WebClient())
```

Comment:
We should not create a new client for each request. This can cause port exhaustion. 
Another simple way is to create a singleton HealthCheckService and initialize the client there. You can move this whole logic in that service